single_cell_RNAseq_snake/bin/de_pool.R

```diff
@@ -0,0 +1,39 @@
+depool_and_save <- function(input_file, output_dir) {
+  if (!file.exists(input_file)) {
+    stop("File not found: ", input_file)
+  }
+
+  # Load the Seurat object
+  seurat_obj <- readRDS(input_file)
+
+  # Determine sample ids
+  seurat_obj$sample_id <- sapply(strsplit(as.character(seurat_obj$BEST.GUESS), ","), function(x) x[1])
```

Comment:
I like this!
I normally do this by putting `strsplit` inside the function(x) part.  But then I need to `strsplit(x, ",")[[1]][1]`.
This is looks cleaner =)